tasks/main.yml

```
@@ -141,41 +139,48 @@
 
 - name: Build Command - Join to a specific Domain Controller
   set_fact:
-    __ad_integration_join_command: |
-      set -euo pipefail
-      echo {{ ad_integration_password | quote }} | realm join -U \
-        {{ ad_integration_user | quote }} --membership-software \
-        {{ ad_integration_membership_software | quote }} \
-        {{ ad_integration_join_parameters }} \
-        {{ ad_integration_join_to_dc | quote }}
+    __ad_integration_join_command: >-
+      realm join -U {{ ad_integration_user | quote }} --membership-software
+      {{ ad_integration_membership_software | quote }}
+      {{ ad_integration_join_parameters }}
+      {{ ad_integration_join_to_dc | quote }}
+    __ad_integration_debug_command: >-
+      realm join -U {{ ad_integration_user | quote }} --membership-software
+      {{ ad_integration_membership_software | quote }}
+      {{ ad_integration_join_to_dc | quote }}
   no_log: true
-  when:
-    - ad_integration_join_to_dc is not none
+  when: ad_integration_join_to_dc is not none
 
 - name: Build Join Command - Perform discovery-based realm join operation
   set_fact:
-    __ad_integration_join_command: |
-      set -euo pipefail
-      echo {{ ad_integration_password | quote }} | realm join -U \
-        {{ ad_integration_user | quote }} --membership-software \
-        {{ ad_integration_membership_software | quote }} \
-        {{ ad_integration_join_parameters }} \
-        {{ ad_integration_realm | quote }}
+    __ad_integration_join_command: >-
+      realm join -U {{ ad_integration_user | quote }} --membership-software
+      {{ ad_integration_membership_software | quote }}
+      {{ ad_integration_join_parameters }}
+      {{ ad_integration_realm | quote }}
+    __ad_integration_debug_command: >-
+      realm join -U {{ ad_integration_user | quote }} --membership-software
+      {{ ad_integration_membership_software | quote }}
+      {{ ad_integration_realm | quote }}
   no_log: true
-  when:
-    - ad_integration_join_to_dc is none
+  when: ad_integration_join_to_dc is none
 
 - name: Show the join command for debug
   debug:
-    msg: "Would run: '{{ __ad_integration_join_command }}'"
+    msg: >-
```

```suggestion
    msg:
      - >-
        Would run the following command. Note that ad_integration_join_parameters
        have been removed for security purposes, the role will pass them to the actual
        realm join command when running wihtout check mode.
      -  "{{ __ad_integration_debug_command }}"
```
Make it a little clearer with the command on a separate line.

---

done